src/main/java/org/mockito/internal/matchers/CapturingMatcher.java

```diff
@@ -26,16 +28,17 @@ public String toString() {
     }
 
     public T getLastValue() {
-        if (arguments.isEmpty()) {
-            throw noArgumentValueWasCaptured();
-        }
-
-        return (T) arguments.getLast();
+        synchronized (arguments) {
```

Comment:
@SimY4 
It doesn't work cause the synchronized List uses its own lock. A ReadWriteLock is a better choice here or a least a dedicated lock-Object, using a List as a lock violates seperation of concerns and should be avoided.